### PR TITLE
Uncertainties shifts branches and disable up/down SVfit

### PR DESCRIPTION
**Pull request by francescobrivio**
Added branches to store Energy Scale uncertainties for JEC, TES and EES. They will be used in KLUB to shift the objects (jets, taus & MET) and re-compute SVfit only on the final selected pairs.
- Added branches to store JEC regrouped uncertainty sources. From twiki: https://twiki.cern.ch/twiki/bin/view/CMS/JECUncertaintySources#Run_2_reduced_set_of_uncertainty
- Added branches to store TES and EES uncertainties
- Disabled SVfit computation for all the up/down variations in order to save computing time
- Up/Down variations for daughters, jets and MET are still save but should be considered only for "global uncertainties" (_i.e._ probably will not be used in the final analysis)

Still to be added: branches for MES uncertainties (mu->tauh fake energy scale)

### CODE REVIEW

File: NtupleProducer/plugins/HTauTauNtuplizer.cc [hunk 1]
```diff
@@ -2032,6 +2137,63 @@ void HTauTauNtuplizer::beginJob(){
...
   myTree->Branch("jets_jecUnc" , &_jets_jecUnc);
+  // JEC Regrouped uncertainty sources
+  myTree->Branch("jets_jetUncRegrouped_FlavorQCD_up"  , &_SourceUncValRegrouped_up["FlavorQCD"]); // up variations
+  myTree->Branch("jets_jetUncRegrouped_RelativeBal_up", &_SourceUncValRegrouped_up["RelativeBal"]);
+  myTree->Branch("jets_jetUncRegrouped_HF_up"         , &_SourceUncValRegrouped_up["HF"]);
+  myTree->Branch("jets_jetUncRegrouped_BBEC1_up"      , &_SourceUncValRegrouped_up["BBEC1"]);
+  myTree->Branch("jets_jetUncRegrouped_EC2_up"        , &_SourceUncValRegrouped_up["EC2"]);
+  myTree->Branch("jets_jetUncRegrouped_Absolute_up"   , &_SourceUncValRegrouped_up["Absolute"]);
+  myTree->Branch("jets_jetUncRegrouped_Total_up"      , &_SourceUncValRegrouped_up["Total"]);
+  myTree->Branch("jets_jetUncRegrouped_FlavorQCD_dw"  , &_SourceUncValRegrouped_dw["FlavorQCD"]); // up variations
```
Comment (camendola) [inline NtupleProducer/plugins/HTauTauNtuplizer.cc]: -> down variations

File: NtupleProducer/plugins/HTauTauNtuplizer.cc [hunk 2]
```diff
@@ -2032,6 +2137,63 @@ void HTauTauNtuplizer::beginJob(){
...
+  myTree->Branch("jets_jetUncRegrouped_Total_dw"      , &_SourceUncValRegrouped_dw["Total"]);
+  if (theYear == 2016)
+  {
+    myTree->Branch("jets_jetUncRegrouped_BBEC1_YEAR_up"         , &_SourceUncValRegrouped_up["BBEC1_2016"]); // up variations
+    myTree->Branch("jets_jetUncRegrouped_EC2_YEAR_up"           , &_SourceUncValRegrouped_up["EC2_2016"]);
+    myTree->Branch("jets_jetUncRegrouped_Absolute_YEAR_up"      , &_SourceUncValRegrouped_up["Absolute_2016"]);
+    myTree->Branch("jets_jetUncRegrouped_HF_YEAR_up"            , &_SourceUncValRegrouped_up["HF_2016"]);
+    myTree->Branch("jets_jetUncRegrouped_RelativeSample_YEAR_up", &_SourceUncValRegrouped_up["RelativeSample_2016"]);
+
+    myTree->Branch("jets_jetUncRegrouped_BBEC1_YEAR_dw"         , &_SourceUncValRegrouped_dw["BBEC1_2016"]); // up variations
```
Comment (camendola) [inline NtupleProducer/plugins/HTauTauNtuplizer.cc]: -> down variations

File: NtupleProducer/plugins/HTauTauNtuplizer.cc [hunk 3]
```diff
@@ -2032,6 +2137,63 @@ void HTauTauNtuplizer::beginJob(){
...
+  }
+  if (theYear == 2017)
+  {
+    myTree->Branch("jets_jetUncRegrouped_BBEC1_YEAR_up"         , &_SourceUncValRegrouped_up["BBEC1_2017"]); // up variations
+    myTree->Branch("jets_jetUncRegrouped_EC2_YEAR_up"           , &_SourceUncValRegrouped_up["EC2_2017"]);
+    myTree->Branch("jets_jetUncRegrouped_Absolute_YEAR_up"      , &_SourceUncValRegrouped_up["Absolute_2017"]);
+    myTree->Branch("jets_jetUncRegrouped_HF_YEAR_up"            , &_SourceUncValRegrouped_up["HF_2017"]);
+    myTree->Branch("jets_jetUncRegrouped_RelativeSample_YEAR_up", &_SourceUncValRegrouped_up["RelativeSample_2017"]);
+
+    myTree->Branch("jets_jetUncRegrouped_BBEC1_YEAR_dw"         , &_SourceUncValRegrouped_dw["BBEC1_2017"]); // up variations
```
Comment (camendola) [inline NtupleProducer/plugins/HTauTauNtuplizer.cc]: -> down variations

File: NtupleProducer/plugins/HTauTauNtuplizer.cc [hunk 4]
```diff
@@ -2032,6 +2137,63 @@ void HTauTauNtuplizer::beginJob(){
...
+  }
+  if (theYear == 2018)
+  {
+    myTree->Branch("jets_jetUncRegrouped_BBEC1_YEAR_up"         , &_SourceUncValRegrouped_up["BBEC1_2018"]); // up variations
+    myTree->Branch("jets_jetUncRegrouped_EC2_YEAR_up"           , &_SourceUncValRegrouped_up["EC2_2018"]);
+    myTree->Branch("jets_jetUncRegrouped_Absolute_YEAR_up"      , &_SourceUncValRegrouped_up["Absolute_2018"]);
+    myTree->Branch("jets_jetUncRegrouped_HF_YEAR_up"            , &_SourceUncValRegrouped_up["HF_2018"]);
+    myTree->Branch("jets_jetUncRegrouped_RelativeSample_YEAR_up", &_SourceUncValRegrouped_up["RelativeSample_2018"]);
+
+    myTree->Branch("jets_jetUncRegrouped_BBEC1_YEAR_dw"         , &_SourceUncValRegrouped_dw["BBEC1_2018"]); // up variations
```
Comment (camendola) [inline NtupleProducer/plugins/HTauTauNtuplizer.cc]: -> down variations

**Comment by francescobrivio** [comment]
Set `TAUCUT=18` so that we store all taus with pT>18 GeV. In this way we do not exclude possible taus with uncorrected pT <20 GeV that could actually be shifted to >20 GeV once we apply TES and EES corrections.
The thresold of `18 GeV` is given by the fact that only taus with pT>18 GeV are stored in miniAOD, as reported in: https://twiki.cern.ch/twiki/bin/viewauth/CMS/TauIDRecommendationForRun2#Kinematic_tau_selection

**Comment by francescobrivio** [comment]
Added muon->tauh fake Energy scale uncertainty branches.
According to tauID twiki [*], there is no need  for muon to tau energy scale correction, so no central correction is applied. Instead, the uncertainties on MES are set to 1% and they are saved in two new branches.

[*] https://twiki.cern.ch/twiki/bin/viewauth/CMS/TauIDRecommendationForRun2#Corrections_to_be_applied_to_AN2

**Comment by francescobrivio** [comment]
@camendola @dzuolo @davignon I think this PR is now complete, if you think it's ok I will merge it

**Comment by francescobrivio** [comment]
I added a new plugin to correct the MET for the central shifts (due to TES and EES) of the taus. Tested on data (results in no shift as expected) and MC (results in the expected shift).
The MET branches are:
- `met` and `metphi`: now contain the **corrected** MET (the one to be used)
-  `met_er`and `met_er_phi`: now contain the **uncorrected** MET (stored just for reference)

**Pull request merged**